lib/pure/os.nim

```diff
@@ -2021,6 +2026,8 @@ iterator walkDir*(dir: string; relative=false): tuple[kind: PathComponent, path:
             let errCode = getLastError()
             if errCode == ERROR_NO_MORE_FILES: break
             else: raiseOSError(errCode.OSErrorCode)
+      elif checkError:
+        raiseOSError(osLastError())
```

Comment:
raiseOSError(osLastError())
=>
raiseOSError(osLastError(), dir)

to show useful runtime info

ditto with other instance of raiseOSError

Reply:
done.